frontend/catalyst/programs/verification.py

```diff
@@ -185,3 +192,57 @@ def _obs_checker(obs):
         _obs_checker(obs)
 
     return (tape,), lambda x: x[0]
+
+
+@transform
+def validate_observables(
```

Comment:
I wonder if the gradient validation of observables shouldn't be moved here, that way observable validation happens in one place rather than two different places

Reply:
I would say that the observable validation does happen in one place, and that one place is device `preprocess` - having separate validation transforms here just splits that into smaller, easier to read functions. I think for code clarity (and similarity to the PennyLane convention), a series of transforms added conditionally to the `TransformProgram` is better than one large transform that performs many different checks depending on conditional inputs. 

We've moved away from that convention for operations specifically because with nested tapes there's a performance concern and we want to minimize how many times we iterate through the operations. But since we don't have a similar concern for observables, I'd rather stick to several smaller transforms added in `preprocess`.